lua/telescope/utils.lua

```diff
@@ -147,6 +148,83 @@ utils.path_tail = (function()
   end
 end)()
 
+utils.slice = function(tbl, s, e)
+  local pos, new = 1, {}
+
+  for i = s, e do
+    new[pos] = tbl[i]
+    pos = pos + 1
+  end
+
+  return new
+end
+
+
+
+PATHS = {}
+utils.path_smart = function(initial_path)
+  local final = initial_path
+  if (table.getn(PATHS) ~= 0) then
+    local diff_counts = { 1, }
+    local dirs = vim.split(initial_path, "/")
+    for _, path in pairs(PATHS) do
+      local _dirs = vim.split(path, "/")
+      local l = table.getn(_dirs)
+      local differs = 0
+      for i = 0, l do
+        if (dirs[i] ~= _dirs[i]) then
+          differs = i
+          break
+        end
+      end
+      table.insert(diff_counts, differs)
+    end
+    local max = math.max(unpack(diff_counts))
+    local length = table.getn(dirs)
+    local final_table = utils.slice(dirs, max, length)
+    if (max == length) then
+      final_table = utils.slice(dirs, length - 1, length)
+    end
+    final = table.concat(final_table, "/")
+  else
+    local length = string.len(final)
+    final = utils.slice(initial_path, length - 1, length)
+    final = table.concat(final, "/")
```

Comment:
If you are only using slice for create a string afterwards you should look for a different method.

Reply:
yeah i figured that but i was like “i’m already using it 🤷🏻‍♀️“. 

it’s a quick hack not production ready at all. i’ll fix these if i’m going to open the pr on the main repo. i appreciate the feedback